### DISCUSSION TITLE
Entity-Relation-Diagram of graph types

### DISCUSSION THREAD

**Discussion by mdre** [ArcadeData/arcadedb#739] [Ideas]
It could be nice to have a way to draw a graph based on types and edges so we can see the abstract graph like we do in a ERD. 

This is a short example of relation in the database. There's 134 different class types.

![imagen](https://user-images.githubusercontent.com/5312780/173203606-6ed1f5fa-2db3-45a0-8a8e-4ccf7b4e50f8.png)

but if I want a DER it should be something like this:

![imagen](https://user-images.githubusercontent.com/5312780/173203626-9931f84a-0dff-4a5c-8d29-472bba06d64c.png)

(I omit the name of relationships in the graph and should be added)
I think that if we can register some metadata in the edge, like (type) in (cardinality) - (cardinality) out (type) , the graph could be made.
and if a edge do not have that info it is ignored in the Graph-Relation-Explain graph
The relations should be visible even if no instance exist.

**Comment by rakesh-tmdc** [comment]
Yes, I agree, this would be great thing if we could have something like information schema to get the ontology of metadata in database.

**Comment by lvca** [comment]
Little update, with https://github.com/ArcadeData/arcadedb/discussions/1026 we'd be able to describe the graph schema and then render it in Studio. I still think another cool feature would be a command that browses the database and build the dynamic schema with data present in the database, no matter if you defined a schema or not.
